sdk/metric/exemplar.go

```diff
@@ -69,7 +69,7 @@ func DefaultExemplarReservoirProviderSelector(agg Aggregation) exemplar.Reservoi
 		//   provided, the default size MAY be the number of possible
 		//   concurrent threads (e.g. number of CPUs) to help reduce
 		//   contention. Otherwise, a default size of 1 SHOULD be used.
-		n = runtime.NumCPU()
+		n = runtime.GOMAXPROCS(0)
```

Comment:
```suggestion
		//
		// Use runtime.GOMAXPROCS instead of runtime.NumCPU to support
		// containerized environments that may have less than the total number
		// of logical CPUs available on the local machine allocated to it.
		n = runtime.GOMAXPROCS(0)
```